### PR TITLE
mgmt, support creating virtual machine with scale set

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Fix https://github.com/Azure/azure-sdk-for-java/issues/27580.
Added `virtualMachineScaleSetId()` and `withExistingVirtualMachineScaleSet()` to `VirtualMachine`.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java [hunk 1]
```diff
@@ -1765,6 +1768,16 @@ WithSystemAssignedIdentityBasedAccessOrCreate withSystemAssignedIdentityBasedAcc
...
         }
 
+        /** The stage of the VM definition allowing to specify virtual machine scale set */
+        interface WithScaleSet {
+            /**
+             * Specifies an existing virtual machine scale set for the virtual machine.
+             * @param scaleSet the virtual machine scale set with flexible orchestration mode
+             * @return the next stage of the definition
+             */
+            WithManagedCreate withExistingVirtualMachineScaleSet(VirtualMachineScaleSet scaleSet);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:1778]: WithManagedCreate or WithCreate  ?

File: sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 - Supported disk encryption set in `Disk` and `VirtualMachine`.
 - Changed to use PATCH for `GalleryImage` update.
 - Supported ephemeral OS disk in `VirtualMachine`.
+- Supported creating `VirtualMachine` with existing `VirtualMachineScaleSet`
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md:10]: nit, period at end

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/CHANGELOG.md]: Added.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java:1778]: WithManagedCreate I believe, I added a assertion in the test: https://github.com/Azure/azure-sdk-for-java/blob/bd048014cd89ee2b89c9bf9749f363a3e40b92dd/sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java#L1377-L1398

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**